### DISCUSSION TITLE
learner/user/v1/signup api working on local setup but giving whitelist error on prod

### DISCUSSION THREAD

**Discussion by paritshivani** [Sunbird-Lern/Community#131] [Issues] [answered]
Hello Team,
Hope you are doing well!..

I am doing some custom flow to user signup in my project
For that I used **/learner/user/v1/signup** API. Its working fine for locahost but giving ** "Forbidden: API WHITELIST Access is denied"** error on production.
Any idea why it happens and whats the solution?
Can anyone help me here?

Working curl:

`curl 'http://localhost:3000/learner/user/v1/signup' \
  -H 'Accept: application/json' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: somecokie' \
  -H 'Origin: http://localhost:3000' \
  -H 'Referer: http://localhost:3000/signup?client_id=portal&state=a1eab6e8-7979-492e-865b-38276ab4fefe&redirect_uri=http:%2F%2Flocalhost:3000%2Fresources%3Registration%26auth_callback%3D1&scope=openid&response_type=code&version=4&error_callback=https:%2F%2Fnulp.niua.org%2Fauth%2Frealms%2Fsunbird%2Fprotocol%2Fopenid-connect%2Fauth' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36' \
  -H 'X-App-Id: local.sunbird.portal' \
  -H 'X-App-Version: 4.4.1' \
  -H 'X-Channel-Id: 0134866036584857600' \
  -H 'X-Device-ID: devId' \
  -H 'X-Org-code: 0134866036584857600' \
  -H 'X-Request-ID: reqId' \
  -H 'X-Session-ID: sessionId' \
  -H 'X-Source: web' \
  -H 'X-msgid: 2b37bf11-2801-cb86-4bd8-5246e43473c0' \
  -H 'sec-ch-ua: "Chromium";v="112", "Google Chrome";v="112", "Not:A-Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'ts: 2023-06-13T15:34:00+05:30' \
  --data-raw '{"params":{"source":"portal","signupType":"self"},"request":{"firstName":"testUser","password":"Password@123","dob":"2020","email":"testUser@yopmail.com","emailVerified":true,"channel":"nulp-sbm","framework":{"state":["Karnataka"],"urbanLocalBody":["test"],"designation":["test"],"id":["framid"]}}}' \
  --compressed`

Non Working curl:
`curl 'https://prodUrl/learner/user/v1/signup' \
  -H 'Accept: application/json' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: somecokie' \
  -H 'Origin: https://prodUrl' \
  -H 'Referer: https://prodUrl/signup?client_id=portal&state=ed8cee47-484a-4915-b414-6eb5d5748098&redirect_uri=https:%2F%2FprodUrl%2Fresources%3Registration%26auth_callback%3D1&scope=openid&response_type=code&version=4&error_callback=https:%2F%2Fnulp.niua.org%2Fauth%2Frealms%2Fsunbird%2Fprotocol%2Fopenid-connect%2Fauth' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36' \
  -H 'X-App-Id: prodUrl.portal' \
  -H 'X-App-Version: 4.4.1' \
  -H 'X-Channel-Id: 0134866036584857600' \
  -H 'X-Device-ID: devId' \
  -H 'X-Org-code: 0134866036584857600' \
  -H 'X-Request-ID: d3fe6834-fecb-d54a-444e-d71fc3246b3c' \
  -H 'X-Session-ID: d5657443-da8c-f41c-845c-80aa04df6aae' \
  -H 'X-Source: web' \
  -H 'X-msgid: d3fe6834-fecb-d54a-444e-d71fc3246b3c' \
  -H 'sec-ch-ua: "Chromium";v="112", "Google Chrome";v="112", "Not:A-Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'ts: 2023-06-13T10:34:03+05:30' \
  --data-raw '{"params":{"source":"portal","signupType":"self"},"request":{"firstName":"Komal Mane","password":"Password@123","dob":"1993","email":"komal_sbm1@yopmail.com","emailVerified":true,"channel":"nulp-sbm","framework":{"state":["test"],"urbanLocalBody":["Test"],"designation":["ttest"],"id":["framidtes"]},"reqData":"{\"iv\":\"a03e290aa09c4930e9d04c589b07457c\",\"encryptedData\":\"1213f2165a75e9e1a7ed5780ab7ef0435b73a05424c8b9c09a5167fa1924b4d2a0dc52c2398da82eb450bdb7236c14af8434289a8445fcde9ffea18fa2c6e7b9\"}"}}' \
  --compressed`

cc: @coolbung, @reshmi-nair , @vinukumar-vs,

**Comment by AmiableAnil** [comment]
@vinukumar-vs @rajeevsathish Can you please look into this.
cc: @rhwarrier @reshmi-nair

**Comment by paritshivani** [comment]
Can anyone help me here?

**Reply by vinukumar-vs** [reply, marked as answer]
@paritshivani You have to use the API manager endpoint when you try in production.  Please check with your devops team to find the API manager end-point to call the sign-up.

Sample endpoint will be : `{{host}}/api/v1/singup`

You can refer the below for Sunbird 
https://github.com/project-sunbird/sunbird-devops/blob/b5797a4039a45c8f2c813ef343529b8a4212a009/ansible/roles/kong-api/defaults/main.yml#L991

@AmiableAnil [@gandham-santhosh](https://github.com/gandham-santhosh) [@rajeevsathish](https://github.com/rajeevsathish)

**Reply by Jayaprakash8887** [reply]
@paritshivani Please mark this ticket as answered if there are no further queries.
